reaktive/src/commonTest/kotlin/com/badoo/reaktive/observable/ObservableByEmitterTest.kt

```diff
@@ -0,0 +1,194 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.disposable.Disposable
+import io.mockk.Ordering
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+class ObservableByEmitterTest {
+
+    private lateinit var disposable: Disposable
+    private lateinit var emitter: ObservableEmitter<Int>
+
+    private val observer =
+        mockk<ObservableObserver<Int>>(relaxed = true) {
+            every { onSubscribe(any()) } answers { disposable = arg(0) }
+        }
+
+    init {
+        observable<Int> { emitter = it }.subscribe(observer)
+    }
+
+    @Test
+    fun onSubscribe_called_WHEN_subscribe() {
+        verify { observer.onSubscribe(any()) }
+    }
+
+    @Test
+    fun emitted_same_values_and_completed_in_the_same_order() {
+        emitter.onNext(1)
+        emitter.onNext(2)
+        emitter.onNext(3)
+        emitter.onComplete()
+
+        verify(ordering = Ordering.ORDERED) {
```

Comment:
let's create TestObserver similar to RxJava TestObserver http://reactivex.io/RxJava/javadoc/io/reactivex/observers/TestObserver.html